### DISCUSSION TITLE
Is ASI Loader's "DontLoadFromDllMain" implemented?

### DISCUSSION THREAD

**Discussion by eddmanx** [elishacloud/dxwrapper#235] [General]
Some of nemesis2000's widescreen fixes require a certain exe size to function, and he's included a patcher to change the exe.

However, I've noticed the fix can be made to work without patching the exe by using ASI Loader's "DontLoadFromDllMain" option.

I tried it with dxwrapper under `[General]` and `[Plugins]`, but it has no effect.

**Comment by elishacloud** [comment]
I guess it could be that the widescreen fix checks the exe size in the DllMain.  I don't know much about how nemesis2000's widescreen fixes work, but if you skip that code then it ignores the exe size and continues to do the rest of the fixes.

However, this may not be safe in two ways.  First, there could be other fixes in the DllMain that are skipped.  Second, some of the fixes may break the game if your using an unsupported exe.

At this point, I have no plans to add a `DontLoadFromDllMain` option.  A better way would be to hack the widescreen fix to bypass the size check.

**Comment by eddmanx** [comment]
So I tested two nemesis WS fixes, Hitman 2 and Metal Gear Solid, with wrong size exe files, using the standalone ASI Loader.

Hitman 2 crashes at launch when `DontLoadFromDllMain` is set to `1`.
(EDIT: the crash happens even without the WS asi in the folder, so it's not related to that.)

MGS launches but the fix doesn't take effect (the image remains 4:3) when `DontLoadFromDllMain` is set to `0`.

Both work when the value is set to the opposite.

Does that indicate anything, considering MGS's fix works when DllMain is **not** used?

EDIT:

With patched, correct size exe, both games work with `DontLoadFromDllMain` set to both `0` and `1`.
